### PR TITLE
Rename the test_listing_fixture to test_processor

### PR DESCRIPTION
**Pull request by mtreinish**
The name test_listing_fixture was a bit confusing. It was a hold over
from test_repository that was never renamed. Even in testrepository the
function of the module is much more than simply listing tests. It's
where all the test workers are launched. While if you look at the
internals of the module the life of the fixture is around the temporary
test list files it creates. But, this name is confusing for people not
super familiar with the internals.

To try and clear up any potential confusion here this commit renames it
to the more generic 'test_processor' (and the Fixture class inside).
This hopefully makes it more clear that the role of the module is around
processing the tests, and more than just listing things.

Closes Issue #21

### CODE REVIEW

**Comment by mtreinish** [comment]
This is going to conflict with my other open pull request https://github.com/mtreinish/stestr/pull/66 which reworks the entire internal python interface. We should merge that much larger change before this one. It'll be much easier to rebase this.

File: stestr/config_file.py [hunk 1]
```diff
@@ -33,19 +33,19 @@ def __init__(self, config_file):
         self.config_file = config_file
 
     def get_run_command(self, options, test_ids=None, regexes=None):
-        """Get a test_listing_fixture.TestListingFixture for this config file
+        """Get a test_processor/.TestProcessorFixture for this config file
```
Comment (mtreinish) [inline stestr/config_file.py]: Nope that was a mistake, I'll fix it in a new revision. I'm going to wait until after #66 merges before I bother though.

Comment (masayukig) [inline stestr/config_file.py]: OK :) I'll look at #66 , anyways.

**Comment by mtreinish** [comment]
I rebased this on current head, which includes #66 now, and also fixed the typo you pointed out

**Comment by masayukig** [comment]
thanks! lgtm

**Pull request merged**